docs/intro/tutorial.rst

```diff
@@ -1,141 +1,446 @@
+item class.
+
+.. _attrs: https://www.attrs.org/en/stable/
+.. _itemadapter: https://github.com/scrapy/itemadapter
+
+Copy the following code into ``tutorial-project/tutorial/items.py``:
+
+.. literalinclude:: /tutorial-project/tutorial/items.py
+   :language: python
+   :lines: 1-6
+
+This code defines a ``Book`` item class, with a single, required, string
+``title`` attribute to store the book title.
+
+``Book`` is a minimal class designed specifically for this tutorial. In real
+web-poet projects, you will usually define item classes with many more
+attributes.
+
+Also mind that, while in this tutorial you use ``Book`` only for data from 1
+website, `books.toscrape.com`_, item classes are usually meant to be usable for
+many different websites that provide data with a similar data schema.
+
+
+Create a page object class
+==========================
+
+To write web data extraction code with web-poet, you write :ref:`page object
+classes <page-objects>`, Python classes that define how to extract data from a
+given type of input, usually some type of webpage from a specific website.
+
+In this tutorial you will write a page object class for webpages of
+`books.toscrape.com`_ that show details about a book, such as these:
+
+-   http://books.toscrape.com/catalogue/the-exiled_247/index.html
+-   http://books.toscrape.com/catalogue/when-we-collided_955/index.html
+-   http://books.toscrape.com/catalogue/set-me-free_988/index.html
+
+Copy the following code into
+``tutorial-project/tutorial/pages/books_toscrape_com.py``:
+
+.. literalinclude:: /tutorial-project/tutorial/pages/books_toscrape_com.py
+   :language: python
+   :lines: 1-11
+
+In the code above:
+
+-   You define a page object class named ``BookPage`` by subclassing
+    :class:`~web_poet.pages.WebPage`.
+
+    It is possible to create a page object class subclassing instead the
+    simpler :class:`~web_poet.pages.ItemPage` class, but
+    :class:`~web_poet.pages.WebPage`:
+
+    -   Indicates that your page object class requires an HTTP response as
+        input, which gets stored in the
+        :attr:`~web_poet.pages.WebPage.response` attribute of your page object
+        class.
+
+    -   Provides attributes like :attr:`~web_poet.pages.WebPage.html` and
+        :attr:`~web_poet.pages.WebPage.url`, and methods like
+        :meth:`~web_poet.pages.WebPage.css`,
+        :meth:`~web_poet.pages.WebPage.urljoin`, and
+        :meth:`~web_poet.pages.WebPage.xpath`, that make it easier to write
+        web data extraction code.
+
+-   ``BookPage`` declares ``Book`` as its return type.
+
+    :class:`~web_poet.pages.WebPage`, like its parent class
+    :class:`~web_poet.pages.ItemPage`, is a :ref:`generic class <generics>`
+    that accepts a type parameter. Unlike most generic classes, however, the
+    specified type parameter is used for more than type hinting: it determines
+    the item class that is used to store the data that fields return.
```

Comment:
Let's also mention that the `.to_item()` method that `WebPage` inherits from `ItemPage` is responsible for returning the said `Book` item.

Reply:
I am unsure about this. It is covered in the following section, in the 3rd bullet point at the end, and I think that may be the best point to cover this.